dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLitmusC.java

```diff
@@ -123,27 +119,31 @@ public Object visitGlobalDeclaratorArray(LitmusCParser.GlobalDeclaratorArrayCont
         Integer size = ctx.DigitSequence() != null ? Integer.parseInt(ctx.DigitSequence().getText()) : null;
 
         if(ctx.initArray() == null && size != null && size > 0){
-            programBuilder.addDeclarationArray(name, Collections.nCopies(size, new IConst(BigInteger.ZERO, -1)));
+            programBuilder.addDeclarationArray(name,size);
             return null;
         }
         if(ctx.initArray() != null){
             if(size == null || ctx.initArray().arrayElement().size() == size){
                 List<IConst> values = new ArrayList<>();
                 for(LitmusCParser.ArrayElementContext elCtx : ctx.initArray().arrayElement()){
                     if(elCtx.constant() != null){
-                        values.add(new IConst(new BigInteger(elCtx.constant().getText()), -1));
+                        values.add(new IValue(new BigInteger(elCtx.constant().getText()), -1));
                     } else {
                         String varName = elCtx.varName().getText();
-                        Address address = programBuilder.getPointer(varName);
+                        Address address = programBuilder.getLocation(varName);
                         if(address != null){
                             values.add(address);
                         } else {
-                            address = programBuilder.getOrCreateLocation(varName).getAddress();
-                            values.add(elCtx.Ast() == null ? address : programBuilder.getInitValue(address));
+                            address = programBuilder.getOrCreateLocation(varName);
+                            values.add(elCtx.Ast() == null ? address : address.getInitialValue(0));
                         }
```

Comment:
Good question: This fragment checks the existence of the location before considering the asterisk.  I am not familiar enough with the semantics of litmus programs.  I would have expected that the name is already defined, and checked for asterisk.

Reply:
`LitmusLinuxTest` succeeds either way.

Reply:
I just run `development` with this code and it works
```
public Object visitGlobalDeclaratorArray(LitmusCParser.GlobalDeclaratorArrayContext ctx) {
        String name = ctx.varName().getText();
        Integer size = ctx.DigitSequence() != null ? Integer.parseInt(ctx.DigitSequence().getText()) : null;

        if(ctx.initArray() == null && size != null && size > 0){
            programBuilder.addDeclarationArray(name, Collections.nCopies(size, new IConst(BigInteger.ZERO, -1)));
            return null;
        }
        if(ctx.initArray() != null){
            if(size == null || ctx.initArray().arrayElement().size() == size){
                List<IConst> values = new ArrayList<>();
                for(LitmusCParser.ArrayElementContext elCtx : ctx.initArray().arrayElement()){
                    if(elCtx.constant() != null){
                        values.add(new IConst(new BigInteger(elCtx.constant().getText()), -1));
                    } else {
                        String varName = elCtx.varName().getText();
                        Address address = programBuilder.getOrCreateLocation(varName).getAddress();
                        values.add(elCtx.Ast() == null ? address : programBuilder.getInitValue(address));
                    }
                }
                programBuilder.addDeclarationArray(name, values);
                return null;
            }
        }
        throw new ParsingException("Invalid syntax near " + ctx.getText());
    }
```

Reply:
> I just run development with this code and it works

[One test](https://github.com/hernanponcedeleon/Dat3M/blob/cad53cc6f51ebd4dfba2e1dbe9a082864cf0134a/dartagnan/src/test/resources/arrays/ok/C-array-ok-17.litmus) in `ArrayValidTest` uses a syntax that this passage was made for.  It currently fails, because it parses `*(&arr1)` to `1`.

Reply:
My bad (I only run the Linux tests). Then we can stick to the original code